---
title: "Thesis Summary"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Objective

Investigate the genetic association between blood plasma proteins and Alzheimerâ€™s Disease (AD) to inform disease prediction and provide signals for functional research

* Identify a shortlist of blood plasma proteins from existing literature that have been associated with AD or AD phenotypes
* Create polygenic risk score (PRS) models for shortlist of proteins and test their association with AD in GERAD, ADNI and ANM samples
* Assess bi-directional genetic association between shortlist of proteins and AD by creating an AD PRS model and testing association with blood plasma protein levels in ANM participants
* Test for causality by conducting Mendelian Randomization (MR) on selected SNPs from blood plasma proteins that are significant in one or both PRS analyses

## Graphical Abstract
![PENDING ACTUAL GRAPHIC](smiley.jpg)

## Results 

### Step 1 - Select target proteins

```{r message=FALSE, warning=FALSE, include=FALSE}
#Load libraries
library(googledrive)
library(googlesheets4)
library(dplyr)
library(knitr)

#log in to google account
drive_auth()
sheets_auth(token = drive_token())

#get protein summary info from google sheet
candidate_proteins <- read_sheet("https://docs.google.com/spreadsheets/d/1J6bPGIw7irreVpgG-cuY-L6nidN6nUVle29XyhE_TZ8/edit#gid=1736283616", sheet = "Table1_Step1_Target_Proteins_For_Results")

target_proteins <- filter(candidate_proteins, In_Final_Shortlist == "Y")
target_proteins <- target_proteins[, 0:4]
target_proteins <- target_proteins %>% mutate("Protein Short Code" = gsub("\\..*","",SOMAmerID)) %>% select(1:2, `Protein Short Code`, everything())

#create table and variable for papers reviewed
#num_papers_reviewed <- nrow(literature_review)
num_candidate_proteins <- nrow(candidate_proteins)
num_target_proteins <- nrow(target_proteins)

```

`r num_candidate_proteins` candidate proteins were identified from a literature review of 33 papers.  
Proteins were marked as candidates if significantly associated with an AD related phenotype in an independent discovery study (univariate or multivariate) or if previously replicated in 3 or more studies in Kiddle et al's comprehensive 2014 review.
 
`r num_target_proteins` proteins with GWAS data from Sun et al were selected for further analysis.
  
######Figure 1: Target proteins for further analysis
```{r message=FALSE, warning=FALSE, echo=FALSE}
#Present table of proteins for analysis
library(kableExtra)
kable(target_proteins) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```
  
            
### Step 2 - Protein PRS -> AD status

```{r message=FALSE, warning=FALSE, include=FALSE}
#Get google sheet with sample data
target_sample_summary <- read_sheet("https://docs.google.com/spreadsheets/d/1JwJrrme5_OUVt4GRcbNJ71m2rK7KR-L6HF-gHeKNiK0/edit#gid=0", sheet = "Table2_Step2_Target_Sample")

protein_base_sample_size <- "3,301"
AD_target_sample_size <- target_sample_summary[3,5]
final_PRS_SNP_num <- "5,210,103"
```

#####Data Preparation

Summary GWAS statistics for the `r num_target_proteins` target proteins were downloaded from the repository provided by the [Cambridge Cardiovascular Epidemiology Unit](http://www.phpc.cam.ac.uk/ceu/proteins/) based on `r protein_base_sample_size` healthy individuals *('Base data')*.  
  
Removed **[X]** bi-allelic and duplicate SNPs resulting in **[X]** SNPs for analysis.  

Raw genotype data from GERAD, ADNI and ANM for `r AD_target_sample_size` individuals accessed via KCL Rosalind High Performance Computing cluster *('Target data')*.  

Consistent QC applied [REQUIRES VALIDATION]  
Samples: European ancestry, non-relatives, no other dementia or mild cognitive impairment, call rate > 98%   
SNPs: MAF >0.01, call rate > 98%, HWE p<10^-5, imputed with INFO quality score >0.7  
  
  
Only SNPs genotyped in all three Target data samples were selected from the Base data SNPs resulting in `r final_PRS_SNP_num` SNPs for consideration in the PRS model.  
  
######Figure 2: Target sample summary
```{r message=FALSE, warning=FALSE, echo=FALSE}
names(target_sample_summary)[1] <- ""
kable(target_sample_summary) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#Get google sheet with heritability data
protein_h2 <- read_sheet("https://docs.google.com/spreadsheets/d/1D8TagzAdtFOX-k5okLfKbmEcmdBpxyH0QkBfqFw_Mqo/edit#gid=165014093", sheet = "heritabilityCheckResults")

average_mean_chi_squared <- mean(protein_h2$Mean_Chi_.2)
average_se <- mean(protein_h2$h2_se)
```
    
#####Protein Heritability

Observed protein heritability (h2) was estimated using linkage disequilibrium score regression (LDSR).  
Average mean chi squared was low at `r average_mean_chi_squared`.  
Results treated as indicative given average SE of `r average_se` and known challenges of applying LDSR to samples <5,000.  
  
######Figure 3: Protein Heritability Estimates

```{r message=FALSE, warning=FALSE, echo=FALSE}
library(ggplot2)
protein_h2 <- protein_h2 %>% mutate("Protein Short Code" = gsub("\\..*","",Protein)) %>% select(1, `Protein Short Code`, everything())
h2_data <- protein_h2 %>% select("Protein Short Code", "h2_no_se", "h2_se")
names(h2_data) <- c("Protein", "h2", "h2_se")

ggplot(data = h2_data) + geom_pointrange(mapping = aes(x=Protein, y=h2, ymin=h2-h2_se, ymax=h2+h2_se)) + geom_hline(yintercept=1, linetype="dashed", color = "red") + geom_hline(yintercept=0, linetype="dashed", color = "red") + theme(axis.text.x=element_text(angle = -90, hjust = 0))


```

Genetic correlation (rg) between proteins was also estimated using LDSR.  
No proteins had an rg of over 0.99 at a p-value less than 0.05.  
[X]% results were NA. [Explain NA in context of LDSR].   
  
######Figure 4: Protein Genetic Correlation Matrix   
  
#####Protein PRS Models    
    
PRS models built for each of the `r num_target_proteins` target proteins at 10 pre-defined p-value thresholds 5e-08 | 1e-05 | 1e-04 | 0.0001 | 0.001 | 0.01 | 0.05 | 0.1 | 0.2 | 0.5 | 1.  
Models applied independently to the three raw genotype samples with and without SNPs from the APOE region and meta-analysed.   
  
```
#PRSice command to generate independent PRS models

Rscript /mnt/lustre/groups/proitsi/Jodie/prs/Programmes/PRSice.R \
      --dir /mnt/lustre/groups/proitsi/Alex/PRS_Outputs_All_Covariates \
      --prsice /mnt/lustre/groups/proitsi/Jodie/prs/Programmes/PRSice_linux \
      --base $PROTEIN_PATH \
      --target $TARGET_FILE \
      --stat BETA \
      --score std \
      --binary-target T \
      --prevalence 0.07 \
      --cov-file $TARGET_COVARIATE_FILE \
      --cov-col PC1,PC2,PC3,PC4,PC5,PC6,PC7,AGE,SEX \
      --bar-levels 5e-8,5e-5,5e-4,0.001,0.01,0.05,0.1,0.2,0.5,1 \
      --fastscore \
      --out /mnt/lustre/groups/proitsi/Alex/PRS_Outputs_All_Covariates/$OUTPUT
      
#Example code for meta-analysis
for (protein in proteins){ 
  print("Select data from each of the 10 thresholds")
    test1<-prsResults %>% filter(Protein == protein, Threshold == 5e-08)
    ...
    test10<-prsResults %>% filter(Protein == protein, Threshold == 1)
     
  print("Run meta analysis on each threshold")
    if(exists("test1")) test1.reml<-rma(yi=Coefficient, sei=Standard.Error, method="REML", data=test1)
    ...
    if(exists("test10")) test10.reml<-rma(yi=Coefficient, sei=Standard.Error, method="REML", data=test10)

```
    
Commentary on results    
  
######Figure 5: Protein PRS Associations with AD (Meta-analysed, No APOE SNPs)

```{r message=FALSE, warning=FALSE, echo=FALSE}

prot_prs_meta <- read_sheet("https://docs.google.com/spreadsheets/d/1mNLTBwvKA17XF69qJw2JqS3upuKkAgWh2pXSDv3m1Yk/edit#gid=885908508", sheet = "prsMetaAnalysisResults_NO_APOE")

#get p-value threshold with lowest p-value for each protein in meta-analysis
prot_prs_meta <- prot_prs_meta %>% mutate("Protein Short Code" = gsub("\\..*","",Protein)) %>% select(1, `Protein Short Code`, everything()) %>% select(-Protein)
names(prot_prs_meta)[1] <- "Protein"
prot_prs_meta_min_p <- prot_prs_meta %>% group_by(Protein) %>% filter(p == min(p))


ggplot(prot_prs_meta_min_p, aes(Protein, P_MinusLog10, label=Threshold, fill=Significant)) + geom_bar(stat = "identity") + geom_hline(aes(yintercept=1.3, linetype="nominal p < 0.05"), color = "red") + expand_limits(y = c(0, 2.5)) + geom_text(vjust=-0.5, size=2) + ylab("-log10 p-value") + scale_linetype_manual(name = "Significance", values = c(2, 2), guide = guide_legend(override.aes = list(color = c("red")))) + labs(caption = "P-value threshold for most significant PRS model displayed above bar for each protein") + theme(axis.text.x=element_text(angle = -90, hjust = 0), plot.caption = element_text(hjust = 0, face= "italic")) + scale_fill_manual( values = c( "Y"="blue", "N"="gray" ), guide = FALSE )

```

######Figure 6: Protein PRS Associations with AD (Per sample, No APOE SNPs)

```{r message=FALSE, warning=FALSE, echo=FALSE}
#get p-values for proteins for each of the individual samples
prot_prs_ind <- read_sheet("https://docs.google.com/spreadsheets/d/1x__FaOVEGs_aZol1Y65vlShIGiSWTmQvjf7Q0lpd4nc/edit#gid=1059518305", sheet = "prsResults_1_NO_APOE")

prot_prs_ind <- prot_prs_ind %>% mutate("Protein Short Code" = gsub("\\..*","",Protein)) %>% select(1, `Protein Short Code`, everything()) %>% select(-Protein)
names(prot_prs_ind)[1] <- "Protein"

prot_prs_ind <- prot_prs_ind %>% mutate("Protein_Sample" = paste(Protein, Sample))

ggplot(prot_prs_ind, aes(Protein, P_MinusLog10, fill=Sample)) + geom_bar(stat = "identity", position=position_dodge()) + geom_hline(aes(yintercept=1.3, linetype="nominal p < 0.05"), color = "red") + theme(axis.text.x=element_text(angle = -90, hjust = 0)) + ylab("-log10 p-value") + xlab("Protein") + scale_linetype_manual(name = "Significance", values = c(2, 2), guide = guide_legend(override.aes = list(color = c("red"))))

```
  
### Step 3 - AD PRS -> protein
  
Summarise QC for base and target data e.g. variants removed, proteins with ANM data available
- display final sample sizes in a table [programmatically reference sample sizes?]

(Optional) Estimate statistical power

Display PRS results
- include snippet of PRSice command
- bar chart (No APOE) comparing all proteins (individual samples and meta-analysis) [load in summary results]
- commentary on results 
  
### Step 4 - Bidirectional MR

Summarise preparation 
- Table of proteins analysed
- Table of SNPs selected for each protein
- Table of SNPs selected for AD

Display MR Results 
- Include MR base code
- summary chart of effect size and p-value for each protein (Protein -> AD, AD -> Protein)
- summary table of results for each protein
- comm
- (bonus) select drop down tables for individual charts on each protein




