---
title: "Thesis Summary"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Objective

Investigate the genetic association between blood plasma proteins and Alzheimerâ€™s Disease (AD) to inform disease prediction and provide signals for functional research

* Identify a shortlist of blood plasma proteins from existing literature that have been associated with AD or AD phenotypes
* Create polygenic risk score (PRS) models for shortlist of proteins and test their association with AD in GERAD, ADNI and ANM consortium samples
* Assess bi-directional genetic association between shortlist of proteins and AD by creating an AD PRS model and testing association with blood plasma protein levels in ANM participants
* Test for causality by conducting Mendelian Randomization (MR) on selected SNPs from blood plasma proteins that are significant in one or both PRS analyses

## Method
![PENDING ACTUAL GRAPHIC](smiley.jpg)

## Results 

### Step 1 - Select target proteins

```{r message=FALSE, warning=FALSE, include=FALSE}
#Load libraries
library(googledrive)
library(googlesheets4)
library(dplyr)
library(knitr)
library(kableExtra)
# library(pander)
# library(xtable)

#log in to google account
drive_auth()
sheets_auth(token = drive_token())

#get Protein sheet
candidate_proteins <- read_sheet("https://docs.google.com/spreadsheets/d/1J6bPGIw7irreVpgG-cuY-L6nidN6nUVle29XyhE_TZ8/edit#gid=1736283616", sheet = "Table1_Target_Proteins_For_Results")

target_proteins <- filter(candidate_proteins, In_Final_Shortlist == "Y")
target_proteins <- target_proteins[, 0:4]
#create table and variable for papers reviewed
#num_papers_reviewed <- nrow(literature_review)
num_candidate_proteins <- nrow(candidate_proteins)
num_target_proteins <- nrow(target_proteins)

```

`r num_candidate_proteins` candidate proteins were identified from a literature review of 33 papers.  
Proteins were marked as candidates if significantly associated with an AD related phenotype in an independent discovery study (univariate or multivariate) or if previously replicated in 3 or more studies in Kiddle et al's comprehensive 2014 review.
 
`r num_target_proteins` proteins with GWAS data from Sun et al were selected for further analysis (see Table 1 below).

####Table 1: Target proteins for further analysis
```{r message=FALSE, warning=FALSE, echo=FALSE}
#Present table of proteins for analysis
kable(target_proteins) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```

### Step 2 - Protein PRS -> AD status

Summarise QC for base and target data e.g. variants removed
- display final sample sizes in a table [programmatically reference sample sizes?]

Estimate statistical power

Display protein heritability results 
- show example python code with ldsc 
- chart comparing heritability for all proteins (scatter with variance) [load in summary results]
- correlation matrix for all proteins [load in summary results]
- short commentary

Display PRS results
- include snippet of PRSice command
- bar chart (No APOE) comparing all proteins (individual samples and meta-analysis) [load in summary results]
- commentary on results 
- (appendix) detailed supplementary table [load in summary results]
- (bonus) select drop down tables for individual charts on each protein 
  -> distribution of PRS scores in sample
  -> bar charts of threshold significance
  
  
### Step 3 - AD PRS -> protein
  
Summarise QC for base and target data e.g. variants removed, proteins with ANM data available
- display final sample sizes in a table [programmatically reference sample sizes?]

Estimate statistical power

Display PRS results
- include snippet of PRSice command
- bar chart (No APOE) comparing all proteins (individual samples and meta-analysis) [load in summary results]
- commentary on results 
- (appendix) detailed supplementary table [load in summary results]
- (bonus) select drop down tables for individual charts on each protein 
  -> distribution of PRS scores in sample
  -> bar charts of threshold significance
  
### Step 4 - Bidirectional MR

Summarise preparation 
- Table of proteins analysed
- Table of SNPs selected for each protein
- Table of SNPs selected for AD

(Optional) Estimate statistical power

Display MR Results 
- Include MR base code
- summary chart of effect size and p-value for each protein (Protein -> AD, AD -> Protein)
- summary table of results for each protein
- comm
- (bonus) select drop down tables for individual charts on each protein




