---
title: "Thesis Summary"
output: html_document
runtime: shiny
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Objective

Investigate the genetic association between blood plasma proteins and Alzheimerâ€™s Disease (AD) to inform disease prediction and provide signals for functional research

* Identify a shortlist of blood plasma proteins from existing literature that have been associated with AD or AD phenotypes
* Create polygenic risk score (PRS) models for shortlist of proteins and test their association with AD in three cohorts (GERAD, ADNI and ANM)
* Assess bi-directional genetic association between shortlist of proteins and AD by creating an AD PRS model and testing association with blood plasma protein levels in ANM participants
* Test for causality by conducting Mendelian Randomization (MR) on selected SNPs from blood plasma proteins that are significant in one or both PRS analyses

## Graphical Abstract
![**PENDING ACTUAL GRAPHIC**](smiley.jpg)

## Results 

### Step 1 - Select target proteins

```{r message=FALSE, warning=FALSE, include=FALSE}
#Load libraries
library(googledrive)
library(googlesheets4)
library(dplyr)
library(knitr)

#log in to google account
drive_auth()
sheets_auth(token = drive_token())

#get protein summary info from google sheet
candidate_proteins <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step1_Target_Proteins")

target_proteins <- filter(candidate_proteins, In_Final_Shortlist == "Y")
target_proteins <- target_proteins[, 0:4]
target_proteins <- target_proteins %>% mutate("Protein Short Code" = gsub("\\..*","",SOMAmerID)) %>% select(1:2, `Protein Short Code`, everything())
target_proteins <- target_proteins %>% arrange(desc(`Number of studies where significant association with AD phenotype`))

#create table and variable for papers reviewed
#num_papers_reviewed <- nrow(literature_review)
num_candidate_proteins <- nrow(candidate_proteins)
num_target_proteins <- nrow(target_proteins)

```

`r num_candidate_proteins` candidate proteins were identified from a literature review of 33 papers.  
Proteins were marked as candidates if significantly associated with an AD related phenotype in an independent discovery study (univariate or multivariate) or if previously replicated in 3 or more studies in Kiddle et al's comprehensive 2014 review.
 
`r num_target_proteins` proteins with GWAS data from Sun et al were selected for further analysis.
  
#####Figure 1: Target proteins for further analysis
```{r message=FALSE, warning=FALSE, echo=FALSE}
#Present table of proteins for analysis
library(kableExtra)
kable(target_proteins) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```
  
            
### Step 2 - Protein PRS -> AD status

```{r message=FALSE, warning=FALSE, include=FALSE}
#Get google sheet with sample data
step2_target_sample_summary <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step2_Target_Sample")

protein_base_sample_size <- "3,301"
AD_target_sample_size <- step2_target_sample_summary[3,5]
final_PRS_SNP_num <- "5,210,103"
```

#####Data Preparation

Summary GWAS statistics for the `r num_target_proteins` target proteins were downloaded from the repository provided by the [Cambridge Cardiovascular Epidemiology Unit](http://www.phpc.cam.ac.uk/ceu/proteins/) based on `r protein_base_sample_size` healthy individuals *('Base data')*.  
  
Removed **[X]** bi-allelic and duplicate SNPs resulting in **[X]** SNPs for analysis.  

Raw genotype data from GERAD, ADNI and ANM for `r AD_target_sample_size` individuals accessed via KCL Rosalind High Performance Computing cluster *('Target data')*.  

Consistent QC applied **[REQUIRES VALIDATION]**  
Samples: European ancestry, non-relatives, no other dementia or mild cognitive impairment, call rate > 98%   
SNPs: MAF >0.01, call rate > 98%, HWE p<10^-5, imputed with INFO quality score >0.7  
  
  
Only SNPs genotyped in all three Target data samples were selected from the Base data SNPs resulting in `r final_PRS_SNP_num` SNPs for consideration in the PRS model.  
  
#####Figure 2: Target sample summary
```{r message=FALSE, warning=FALSE, echo=FALSE}
names(step2_target_sample_summary)[1] <- ""
kable(step2_target_sample_summary) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```

```{r message=FALSE, warning=FALSE, echo=FALSE}
#Get google sheet with heritability data
protein_h2 <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step2_Heritability_Check")

average_mean_chi_squared <- mean(protein_h2$Mean_Chi_.2)
average_se <- mean(protein_h2$h2_se)
```
    
#####Protein Heritability

Observed protein heritability (h2) was estimated using linkage disequilibrium score regression (LDSR).  
Average mean chi squared was low at `r average_mean_chi_squared`.  
Results treated as indicative given average SE of `r average_se` and known challenges of applying LDSR to samples <5,000.  
  
#####Figure 3: Protein Heritability Estimates

```{r message=FALSE, warning=FALSE, echo=FALSE, out.width='85%'}
library(ggplot2)
protein_h2 <- protein_h2 %>% mutate("Protein Short Code" = gsub("\\..*","",Protein)) %>% select(1, `Protein Short Code`, everything())
h2_data <- protein_h2 %>% select("Protein Short Code", "h2_no_se", "h2_se")
names(h2_data) <- c("Protein", "h2", "h2_se")

ggplot(data = h2_data) + geom_pointrange(mapping = aes(x=Protein, y=h2, ymin=h2-h2_se, ymax=h2+h2_se)) + geom_hline(yintercept=1, linetype="dashed", color = "red") + geom_hline(yintercept=0, linetype="dashed", color = "red") + theme(axis.text.x=element_text(angle = -90, hjust = 0))


```

Genetic correlation (rg) between proteins was also estimated using LDSR.  
No proteins had an rg of over 0.99 at a p-value less than 0.05.  
*[X]*% results were NA. *[Explain NA in context of LDSR]*.   
  
#####Figure 4: Protein Genetic Correlation Matrix  

**[TABLE TO BE ADDED]**
  
#####Protein PRS Models    
    
PRS models built for each of the `r num_target_proteins` target proteins at 10 pre-defined p-value thresholds 5e-08 | 1e-05 | 1e-04 | 0.0001 | 0.001 | 0.01 | 0.05 | 0.1 | 0.2 | 0.5 | 1.  
Models applied independently to the three raw genotype samples with and without SNPs from the APOE region and meta-analysed.   
Age, sex and 7 principal components for population stratification included as covariates. 
  
```
#PRSice command to generate independent PRS models

Rscript /mnt/lustre/groups/proitsi/Jodie/prs/Programmes/PRSice.R \
      --dir /mnt/lustre/groups/proitsi/Alex/PRS_Outputs_All_Covariates \
      --prsice /mnt/lustre/groups/proitsi/Jodie/prs/Programmes/PRSice_linux \
      --base $PROTEIN_PATH \
      --target $TARGET_FILE \
      --stat BETA \
      --score std \
      --binary-target T \
      --prevalence 0.07 \
      --cov-file $TARGET_COVARIATE_FILE \
      --cov-col PC1,PC2,PC3,PC4,PC5,PC6,PC7,AGE,SEX \
      --bar-levels 5e-8,5e-5,5e-4,0.001,0.01,0.05,0.1,0.2,0.5,1 \
      --fastscore \
      --out /mnt/lustre/groups/proitsi/Alex/PRS_Outputs_All_Covariates/$OUTPUT
      
#Example code for meta-analysis
for (protein in proteins){ 
  print("Select data from each of the 10 thresholds")
    test1<-prsResults %>% filter(Protein == protein, Threshold == 5e-08)
    ...
    test10<-prsResults %>% filter(Protein == protein, Threshold == 1)
     
  print("Run meta analysis on each threshold")
    if(exists("test1")) test1.reml<-rma(yi=Coefficient, sei=Standard.Error, method="REML", data=test1)
    ...
    if(exists("test10")) test10.reml<-rma(yi=Coefficient, sei=Standard.Error, method="REML", data=test10)

```
    
6 proteins were nominally significant (p = <0.05) in the with APOE sample after meta-analysis (APOE, Serum amyloid P component, Complement component C6, Inter-alpha-trypsin inhibitor heavy chain H1, Prostate-specific antigen, Plasma protease C1 inhibitor).

When *[X]* SNPs in the APOE region were removed from the base and target data to control for the known association with APOE, 6 proteins were nominally significant with Insulin-like growth factor-binding protein 2 replacing APOE. 

#####Figure 5a: Proteins with Protein PRS significantly associated with AD
```{r message=FALSE, warning=FALSE, echo=FALSE}
step2_sig_proteins <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step_2_PRS_Sig_Proteins")
kable(step2_sig_proteins) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```


#####Figure 5b: Protein PRS Associations with AD (Meta-analysed, No APOE SNPs)

```{r message=FALSE, warning=FALSE, echo=FALSE, out.width='85%'}

prot_prs_meta <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step2_PRS_Meta_Results_NO_APOE")

#get p-value threshold with lowest p-value for each protein in meta-analysis
prot_prs_meta <- prot_prs_meta %>% mutate("Protein Short Code" = gsub("\\..*","",Protein)) %>% select(1, `Protein Short Code`, everything()) %>% select(-Protein)
names(prot_prs_meta)[1] <- "Protein"
prot_prs_meta_min_p <- prot_prs_meta %>% group_by(Protein) %>% filter(p == min(p))


ggplot(prot_prs_meta_min_p, aes(Protein, P_MinusLog10, label=Threshold, fill=Significant)) + geom_bar(stat = "identity") + geom_hline(aes(yintercept=1.3, linetype="nominal p < 0.05"), color = "red") + expand_limits(y = c(0, 2.5)) + geom_text(vjust=-0.5, size=2) + ylab("-log10 p-value") + scale_linetype_manual(name = "Significance", values = c(2, 2), guide = guide_legend(override.aes = list(color = c("red")))) + labs(caption = "P-value threshold for most significant PRS model displayed above bar for each protein") + theme(axis.text.x=element_text(angle = -90, hjust = 0), plot.caption = element_text(hjust = 0, face= "italic")) + scale_fill_manual( values = c( "Y"="blue", "N"="gray" ), guide = FALSE )

```

#####Figure 6: Protein PRS Associations with AD (Per sample, No APOE SNPs)

```{r message=FALSE, warning=FALSE, echo=FALSE, out.width='85%'}
#get p-values for proteins for each of the individual samples
prot_prs_ind <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step2_PRS_Cohort_Results_NO_APOE")

prot_prs_ind <- prot_prs_ind %>% mutate("Protein Short Code" = gsub("\\..*","",Protein)) %>% select(1, `Protein Short Code`, everything()) %>% select(-Protein)
names(prot_prs_ind)[1] <- "Protein"

prot_prs_ind <- prot_prs_ind %>% mutate("Protein_Sample" = paste(Protein, Sample))

ggplot(prot_prs_ind, aes(Protein, P_MinusLog10, fill=Sample)) + geom_bar(stat = "identity", position=position_dodge()) + geom_hline(aes(yintercept=1.3, linetype="nominal p < 0.05"), color = "red") + theme(axis.text.x=element_text(angle = -90, hjust = 0)) + ylab("-log10 p-value") + xlab("Protein") + scale_linetype_manual(name = "Significance", values = c(2, 2), guide = guide_legend(override.aes = list(color = c("red"))))

```
  
### Step 3 - AD PRS -> protein

```{r message=FALSE, warning=FALSE, echo=FALSE}
#get AD PRS results
ad_prs_res <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step3_PRS_Results_NO_APOE")

ad_prs_res <- ad_prs_res %>% mutate("Protein Short Code" = gsub("\\..*","",Protein)) %>% select(1, `Protein Short Code`, everything()) %>% select(-Protein)
names(ad_prs_res)[1] <- "Protein"

num_ad_prs_proteins <- nrow(ad_prs_res)

```

#####AD PRS Models    
    
PRS model built for AD using Kunkle et al GWAS summary statistics and tested for association with the `r num_ad_prs_proteins` proteins which had blood plasma protein levels collected in the ANM cohort.   
For pancreatic prohormone, amyloid-beta A4 precursor protein-binding family B member 3, fibulin-1, inter-alpha-trypsin inhibitor heavy chain H1 and vitronectin no protein level data available.

#####Figure 7: ANM target sample summary
**[TABLE TO BE COMPLETED]**
```{r message=FALSE, warning=FALSE, echo=FALSE}
step3_target_sample_summary <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step3_Target_Sample")
names(step3_target_sample_summary)[1] <- ""
kable(step3_target_sample_summary) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```

AD PRS models were tested with and without SNPs from the APOE region at 10 pre-defined p-value thresholds 5e-08 | 1e-05 | 1e-04 | 0.0001 | 0.001 | 0.01 | 0.05 | 0.1 | 0.2 | 0.5 | 1.    
No covariates included as protein phenotype created by taking residuals from regression of protein levels with age, gender and centre.

**ANALYSIS TO-DOS**  

+ Test for collinearity between AD PRS score and AD status in ANM and deciles analysis
+ Confirm covariates and run with and without case / control as covariates
+ Run AD and Protein PRS stratified by age
+ Run Protein PRS deciles association analysis with protein levels in ANM cohort (to check level of association with protein)
+ Run Protein PRS deciles association analysis with AD for significant proteins

```
#PRSice command to generate AD PRS models

Rscript /mnt/lustre/groups/proitsi/Jodie/prs/Programmes/PRSice.R \
--dir /mnt/lustre/groups/proitsi/Alex/AD_PRS_to_Protein/PRS_Outputs \
--prsice /mnt/lustre/groups/proitsi/Jodie/prs/Programmes/PRSice_linux \
--base /mnt/lustre/groups/proitsi/Alex/AD_PRS_to_Protein/Base_Data/Post_QC/Kunkle_Stage1_post_qc.txt \
--target /mnt/lustre/groups/proitsi/Alex/AD_PRS_to_Protein/Target_Data/three_batches_imputed.pQTL.QC \
--chr Chromosome \
--bp Position \
--snp  MarkerName \
--A1 Effect_allele \
--A2 Non_Effect_allele \
--stat Beta \
--pvalue Pvalue \
--score std \
--binary-target F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F,F \
--pheno /mnt/lustre/groups/proitsi/Alex/AD_PRS_to_Protein/Target_Data/SOMA_Pheno_Oct19 \
--pheno-col $TARGET_PROTEINS \
--bar-levels 5e-8,5e-5,5e-4,0.001,0.01,0.05,0.1,0.2,0.5,1 \
--fastscore \
--out /mnt/lustre/groups/proitsi/Alex/AD_PRS_to_Protein/PRS_Outputs/Kunkle_AD_to_ANM_WITH_APOE
```

2 proteins were nominally significant (p = <0.05) in the with APOE sample (APOE and Haptoglobin).

When *[X]* SNPs in the APOE region were removed from the base and target data to control for the known association with APOE, 2 proteins were nominally significant with Interleukin-10 replacing APOE. 

#####Figure 8a: Proteins significantly associated with AD PRS
```{r message=FALSE, warning=FALSE, echo=FALSE}
step3_sig_proteins <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step_3_PRS_Sig_Proteins")
kable(step3_sig_proteins) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```

#####Figure 8b: AD PRS Associations with blood plasma protein levels (No APOE SNPs)

```{r message=FALSE, warning=FALSE, echo=FALSE, out.width='85%'}

#convert p-value to minuslog10 p-value and add Significant
ad_prs_res <- ad_prs_res %>% mutate(P_MinusLog10 = -log10(P), Significant = if_else(P < 0.05, "Y","N"))

ggplot(ad_prs_res, aes(Protein, P_MinusLog10, label=Threshold, fill=Significant)) + geom_bar(stat = "identity") + geom_hline(aes(yintercept=1.3, linetype="nominal p < 0.05"), color = "red") + expand_limits(y = c(0, 2.5)) + geom_text(vjust=-0.5, size=2) + ylab("-log10 p-value") + scale_linetype_manual(name = "Significance", values = c(2, 2), guide = guide_legend(override.aes = list(color = c("red")))) + labs(caption = "P-value threshold for most significant PRS model displayed above bar for each protein") + theme(axis.text.x=element_text(angle = -90, hjust = 0), plot.caption = element_text(hjust = 0, face= "italic")) + scale_fill_manual( values = c( "Y"="blue", "N"="gray" ), guide = FALSE )

```
  
### Step 4 - Bidirectional MR

Proteins that were significant in one or both PRS were considered for Mendelian Randomisation (MR).   
This resulted in 8 proteins, excluding APOE.
4 proteins had SNPs that passed criteria for use as genetic instruments:  

+ Genome wide significant cis-eQTLs associated with gene encoding target protein
+ Not associated with confounders
+ Not associated with outcome (except through exposure)
+ Not in LD with other instrumental variables  

**ANALYSIS TO-DOS**  

+ Review constraints to include more SNPs for existing 4 proteins and other 4  
+ Apply statistical biological screening for AD and protein SNP instruments

#####Figure 9: Proteins for MR
```{r message=FALSE, warning=FALSE, echo=FALSE}
step4_mr_proteins <- read_sheet("https://docs.google.com/spreadsheets/d/1N_Wh36t76ps2jK1WJZiFynTCKJsKVnn_4e7jZNZNnRE/edit#gid=493626447", sheet = "Step_4_MR_Proteins")
kable(step4_mr_proteins) %>% kable_styling(bootstrap_options = "condensed", full_width = F, position="left")
```

 
Proteins were analysed individually as an exposure with AD as the outcome. AD was then used as an exposure with each protein as the outcome to test for reverse causality.  
Inverse variance weighted MR and sensitivity analysis (e.g. weighted median, weighted mode, MR Egger) was conducted using the R package [MR Base](https://mrcieu.github.io/TwoSampleMR/). 
 
#####Figure 10: MR Results

<!-- For knitting and publishing to web, use iframe with "runtime: shiny"" removed to allow "knit" and creation of .html file and eval=FALSE-->
<!-- <iframe src="https://alexhandy1.shinyapps.io/mr_results/" width=1000 height=1000" frameBorder="0"></iframe> -->

```{r, echo=FALSE, eval=TRUE}
shinyApp(

  ui <- fluidPage(
     
     titlePanel(""),
    
     sidebarLayout(
        sidebarPanel(
          selectInput("exposure", strong("Exposure"), 
                      choices = c("HP","ITIH1","C1","SAP", "AD"), selected = "HP"),
          selectInput("outcome", strong("Outcome"), 
                      choices = c("HP","ITIH1","C1","SAP", "AD"), selected = "AD")
        ),
        
        mainPanel(
          imageOutput("ScatterPlot"),
          imageOutput("ForestPlot")
        )
     )
  ),
  
  server <- function(input, output) {
     output$ScatterPlot <- renderImage({
       
       if(input$exposure == "AD" && input$outcome != "AD") {
         imagePath1 <- paste("www/AD_Prt_",input$outcome,"_1.png", sep="")
       } else if(input$outcome == "AD" && input$exposure != "AD") {
         imagePath1 <- paste("www/Prt_AD_",input$exposure,"_1.png", sep="")
       } else {
         imagePath1 <- "www/404.png"
       }
  
       list(src = imagePath1, width='85%', height='85%')
     }, deleteFile = FALSE)
     
     output$ForestPlot <- renderImage({
       
       if(input$exposure == "AD" && input$outcome != "AD") {
         imagePath2 <- paste("www/AD_Prt_",input$outcome,"_2.png", sep="")
       } else if(input$outcome == "AD" && input$exposure != "AD") {
         imagePath2 <- paste("www/Prt_AD_",input$exposure,"_2.png", sep="")
       } else {
         imagePath2 <- "www/404.png"
       }
       
       list(src = imagePath2, width='100%', height='100%')
     }, deleteFile = FALSE)
  },

  options = list(height = 500)
)
```


